src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx

```diff
@@ -145,8 +145,11 @@ export default class IntegrationRepos extends AsyncComponent {
         </DropdownButton>
       );
     }
-    let repositories = this.state.integrationRepos.repos;
-    let items = (repositories || []).map(repo => {
+    const repositories = new Set(this.state.itemList.map(i => i.name));
+    let repositoryOptions = this.state.integrationRepos.repos.filter(repo => {
```

Comment:
* Let's make `let`/`const` consistent here
* nit: one liner for filter e.g. `repo => !repositories.has(repo.identifier)`
* You'll want to guard `repos` not being an array like before e.g. `(this.state.integrationRepos.repos || [])`